Runtimes/Supplemental/Distributed/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+cmake_minimum_required(VERSION 3.29)
+
+if($ENV{BUILD_NUMBER})
+  math(EXPR BUILD_NUMBER "$ENV{BUILD_NUMBER} % 65535")
+  set(BUILD_NUMBER ".${BUILD_NUMBER}")
+endif()
+
+project(SwiftDistributed
+  LANGUAGES C CXX Swift 
+  VERSION 6.3.0${BUILD_NUMBER})
+
+if(NOT PROJECT_IS_TOP_LEVEL)
+  message(SEND_ERROR "Swift Distributed must build as a standalone project")
+endif()
+
+set(CMAKE_POSITION_INDEPENDENT_CODE YES)
+set(CMAKE_C_VISIBILITY_PRESET "hidden")
+set(CMAKE_CXX_VISIBILITY_PRESET "hidden")
+set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake/modules")
+
+set(CMAKE_Swift_LANGUAGE_VERSION 5)
+
+set(${PROJECT_NAME}_SWIFTC_SOURCE_DIR
+  "${PROJECT_SOURCE_DIR}/../../../"
+  CACHE FILEPATH "Path to the root source directory of the Swift compiler")
+
+set(${PROJECT_NAME}_VENDOR_MODULE_DIR "${CMAKE_SOURCE_DIR}/../cmake/modules/vendor"
+  CACHE FILEPATH "Location for private build system extension")
+
+include(CxxStandard)
+include(SwiftCallingConventions)
+include(AvailabilityMacros)
+include(ResourceEmbedding)
+include(PlatformInfo)
+
+find_package(SwiftCore)
+
+include("${${PROJECT_NAME}_VENDOR_MODULE_DIR}/Settings.cmake" OPTIONAL)
+
+option(${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION "Generate ABI resilient runtime libraries"
+  ${SwiftCore_ENABLE_LIBRARY_EVOLUTION})
+
+option(${PROJECT_NAME}_ENABLE_PRESPECIALIZATION "Enable generic metadata prespecialization"
+  ${SwiftCore_ENABLE_PRESPECIALIZATION})
+
+include(CatalystSupport)
+include(EmitSwiftInterface)
+include(InstallSwiftInterface)
+
+configure_file("CMakeConfig.h.in"
+  "${PROJECT_BINARY_DIR}/include/swift/Runtime/CMakeConfig.h"
+  ESCAPE_QUOTES @ONLY)
```

Comment:
This seems sketchy. We shouldn't be re-generating the configuration for the runtime build, but should distribute it or pull it from the build tree.

Reply:
Let me see what I can do -- possibly a find module may help here, need to think about the distribution scenario.

Reply:
If we install/export this from Core, `find_package(SwiftCore)` should be able to wire this up.

Reply:
Once the Core and supplemental libraries are built, the `CMakeConfig.h` file aren't used by anything so it shouldn't be necessary so installing it isn't great.

You're right that it's sketchy if the projects get out of sync. I do think that the `SwiftCoreConfig.cmake` should export the appropriate settings and that the supplemental libraries should import them to ensure that the values are kept in sync.

Reply:
I think that for the time being we can control that from the outside then; I don't think that we should regenerate it (point into the build tree with `CMAKE_CXX_FLAGS`).